var/spack/repos/builtin/packages/py-astroid/package.py

```diff
@@ -22,10 +23,15 @@ class PyAstroid(PythonPackage):
     version('1.4.2', sha256='f9007d651f4b3514ea5812127677a4bb681ff194164290cea358987920f24ee6')
     version('1.4.1', sha256='f1ab3ee6f17f9d30981399a52b56a7a7d2747ba24f0aa504e411ee6205a01fc0')
 
+    depends_on('python@3.5:', when='@2.3.3:')
```

Comment:
```suggestion
    depends_on('python@3.5:', when='@2.3.3:', type=('build', 'run'))
```
It looks like 2.0+ requires Python 3.4+ if you want to add that too. 1.X is required for Python 2 support